crates/typst-library/src/visualize/image.rs

```diff
@@ -64,6 +68,60 @@ pub struct ImageElem {
     pub fit: ImageFit,
 }
 
+/// A raster or vector graphic loaded from string.
+///
+/// Supported formats are PNG, JPEG, GIF and SVG.
+///
+/// _Note:_ Work on SVG export is ongoing and there might be visual inaccuracies
+/// in the resulting PDF. Make sure to double-check embedded SVG images. If you
+/// have an issue, also feel free to report it on [GitHub][gh-svg].
+///
+/// ## Example { #example }
+/// ```example
+/// #figure(
+///   image.decode(read("diagram.svg").replace("yellow", "green"), "svg", width: 80%),
+///   caption: [
+///     Results of the molecular testing pipeline.
+///   ],
+/// )
+/// ```
+///
+/// [gh-svg]: https://github.com/typst/typst/issues?q=is%3Aopen+is%3Aissue+label%3Asvg
+///
+/// Display: Image
+/// Category: visualize
+#[func]
+pub fn image_decode(
+    /// Image content
+    content: EcoString,
```

Comment:
Should allow both `Str` and `Bytes` (now that bytes are available). There is already such an enum for the `read` function `data.rs`, you can reuse it here.

I would tend to rename `content` to `data`. The content and format arguments could maybe have a bit more documentation, currently it says nothing the name doesn't already say. Please also terminate each doc comment with a period.